cypress/e2e/po/edit/machine-deployments.po.ts

```diff
@@ -0,0 +1,22 @@
+import PagePo from '@/cypress/e2e/po/pages/page.po';
+import AsyncButtonPo from '@/cypress/e2e/po/components/async-button.po';
+
+export default class MachineDeploymentsCreateEditPo extends PagePo {
+  private static createPath(clusterId: string, nsName?: string, machineSetName?: string ) {
+    const root = `/c/${ clusterId }/manager/cluster.x-k8s.io.machinedeployment`;
+
+    return nsName && machineSetName ? `${ root }/${ nsName }/${ machineSetName }` : `${ root }/create`;
+  }
+
+  static goTo(path: string): Cypress.Chainable<Cypress.AUTWindow> {
+    throw new Error('invalid');
+  }
+
+  constructor(clusterId = 'local', nsName?: string, machineSetName?: string) {
+    super(MachineDeploymentsCreateEditPo.createPath(clusterId, nsName, machineSetName));
+  }
+
+  saveCreateForm(): AsyncButtonPo {
```

Comment:
In term terms of component hierarchy it goes ResourceDetail --> ResourceYaml --> Footer (which contains the button).

So would suggest either creating a Footer PO (containing the button) and adding it to ResourceYamlPo, or just adding this function to ResourceYamlPo. In both cases you would access them via  `ResourceDetailPo` `.resourceYaml`

This pattern also applies to the other machine resources.

Reply:
makes sense now, thanks for clarifying that hierarchy!